README.md

```diff
@@ -412,6 +412,30 @@ need a shell or related tools, use the Alpine or Buster images listed above.
 
 [distroless]: https://github.com/GoogleContainerTools/distroless
 
+### Working with Docker and the Proxy
+
+The containers have the proxy as an `ENTRYPOINT` so, to use the proxy from a container, all you need to do is specify options using the command, and expose the proxy's internal port to the host. For example, you can use:
+
+```bash
+docker run --publish <host-port>:<proxy-port> gcr.io/cloud-sql-connectors/cloud-sql-proxy:2.8.2 --address "0.0.0.0" --port <proxy-port> <instance-connection-name>
+```
+
+You'll need the `--address "0.0.0.0"` so that the proxy doesn't only listen for connections originating from *within* the container.
+
+You will need to authenticate using one of the methods outlined in the [credentials](#credentials) section. If using a credentials file you must mount the file and ensure that the non-root user that runs the proxy has *read access* to the file. These alternatives might help:
+
+1. Change the group of your local file and add read permissions to the group with `chgrp 65532 key.json && chmod g+r key.json`.
+2. If you can't control your file's group, you can directly change the public permissions of your file by doing `chmod o+r key.json`.
+
+    > [!WARNING]
+    > This can be insecure because it allows any user in the host system to read the credential file, which they can use to authenticate to services in GCP.
+
+For example, a full command using a JSON credentials file might look like
+
+```bash
+docker run --publish <host-port>:<proxy-port> --mount type=bind,source="$(pwd)"/sa.json,target=/config/sa.json gcr.io/cloud-sql-connectors/cloud-sql-proxy:2.8.2 --address 0.0.0.0 --port <proxy-port> --credentials-file /config/sa.json <instance-connection-name>
+```
```

Comment:
@enocom I thing long-lived service account keys are slowly trying to be phased out in documentation for other methods like workload identity etc. Is there a way we can showcase workload identity here instead?

Reply:
We're trying to go with Workload Identity for a GH Action in a project I'm in. Coincidentally, I was going to tackle this tomorrow so, if it works and @enocom thinks it's best, I can give it a shot with the Proxy and change the `README.md`.

Reply:
Yeah I'd be curious to hear how that goes. As an aside, we might want to mention here that if you run this in Cloud Shell, or a GCE VM, the auth library will find the token for you, and makes this much simpler.

For on-prem, it's a bit tricky: either WIF or a SA key is necessary.

Reply:
We were able to use Workload Identity by adding this step to our GitHub Action Job:

```yaml
      - id: auth
        name: Auth with Google Cloud
        uses: "google-github-actions/auth@v2"
        with:
          project_id: ${{ vars.PROJECT_ID }}
          workload_identity_provider: ${{ secrets.WORKLOAD_IDENTITY_PROVIDER }}
```

which provides the `steps.auth.outputs.auth_token` output with the token needed to authorize.

It seems the proxy supports either a SA JSON or Application Default Credentials from `gcloud`, so how would this work from within the container? [[ref](https://cloud.google.com/docs/authentication/provide-credentials-adc#local-dev)] All I can think of is requesting a token and injecting it to the container via the `--token` flag, but that sounds a bit more complicated.

Reply:
No pressure but, to clarify, I'm awaiting some guidance on how to use WIF with the proxy container to try it out and document it :).

Reply:
Sorry for the slow response here. If you're OK with a short-lived workload (e.g. CI/CD), then using the token flag would probably be best.

To take a step back, are we solving for GitHub Actions here, or trying to be more generic about giving advice on using the Proxy in an on-prem containerized situation?

Reply:
> are we solving for GitHub Actions here, or trying to be more generic about giving advice on using the Proxy in an on-prem containerized situation?

I think "generic advice". At least my usage is local only for now, so I haven't tried this in CI/CD.

Reply:
I think for the purpose of this PR, the current context is probably adequate.